### ISSUE TITLE
Can AutoFlushWrapper be configured to not flush on config reload/shutdown?

### ISSUE THREAD

**Issue by ben-wallis** [NLog/NLog#3512]
**NLog version**: 4.6.5

**Platform**: .NET 4.7.2

**Current NLog config**

```xml
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="false"
      internalLogLevel="Off" internalLogFile="c:\temp\nlog-internal.log">

  <targets>
     <target xsi:type="File" name="fileTarget" fileName="log_${cached:${date:format=yyyy-MM-dd HH_mm_ss}}.log" layout="${longdate}|${level:uppercase=true}|${logger}|${message} ${exception:format=ToString,StackTrace}" />		
		
     <target name="crashTraceTarget" xsi:type="AutoFlushWrapper" condition="level >= LogLevel.Fatal">
        <target xsi:type="BufferingWrapper" bufferSize="500" overflowAction="Discard">
           <target xsi:type="File" name="crashTraceFileTarget" fileName="crashtrace_${cached:${date:format=yyyy-MM-dd HH_mm_ss}}.log" layout="${longdate}|${level:uppercase=true}|${logger}|${message} ${exception:format=ToString,StackTrace}" />
        </target>
     </target>
  </targets>

  <rules>
    <logger name="*" minlevel="Info" writeTo="fileTarget" />
    <logger name="*" minlevel="Trace" writeTo="crashTraceTarget" />
  </rules>
</nlog>
```

I'm trying to use `AutoFlushWrapper` to automatically write a log file containing the last 500 `>= Trace` logs when a `Fatal` log is written. This works, however the log file is also flushed on config reload and application exit which is undesirable. Is there a way to set this up so that the `AutoFlushWrapper` does not flush on config reload or application exit, i.e. the buffered logs are discarded on clean exit?

I've had a look at the documentation for custom extensions/wrappers but can't see anything that would allow suppressing the flush on config reload/application exit.

**Comment by snakefoot** [comment]
Create #3513 with new property `IgnoreExplicitFlush`

**Comment by 304NotModified** [comment]
Fixed with #3513  :)

**Comment by ben-wallis** [comment]
Perfect, thanks for the quick implementation!
